### DISCUSSION TITLE
Is there any way to change the aggregation temporality according to the metric name (or other label)?

### DISCUSSION THREAD

**Discussion by huange7** [open-telemetry/opentelemetry-java-instrumentation#12549] [Q&A]
When I set 'otel.exporter.otlp.metrics.temporality.preference=delta', the aggregation temporality is set according to the instrument type. I want to manually specify it according to the metric name or when creating a new metric. Is this allowed?

Any reply is appreciated. Thank you.

**Comment by laurit** [comment]
> When I set 'otel.exporter.otlp.metrics.temporality.preference=delta', the aggregation temporality is set according to the instrument type. I want to manually specify it according to the metric name or when creating a new metric. Is this allowed?

I'd say no, I think this might require spec change, though not really my area of expertise so could be wrong. You already asked this question once which I moved to `opentelemetry-java` where there would be a better chance of getting this answered. Since you closed the discussion there I'm not going to move this again. Feel free to open a new discussion in `opentelemetry-java` yourself.

**Reply by huange7** [reply]
thanks for ur reply! 
Sorry. I thought that I had opened the discussion in the wrong repository. I will reopen it.
